program/rust/src/accounts/score.rs

```diff
@@ -0,0 +1,181 @@
+use {
+    super::{
+        AccountHeader,
+        PythAccount,
+    },
+    crate::c_oracle_header::{
+        PC_ACCTYPE_SCORE,
+        PC_MAX_PUBLISHERS,
+        PC_MAX_SYMBOLS,
+        PC_MAX_SYMBOLS_64,
+    },
+    bytemuck::{
+        Pod,
+        Zeroable,
+    },
+    solana_program::{
+        program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    std::cmp::max,
+};
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct PublisherScoresAccount {
+    pub header:         AccountHeader,
+    pub num_publishers: usize,
+    pub num_symbols:    usize,
+    // a constant used to normalize the scores
+    pub z:              u32,
+
+    // array[x][y] is a u64 whose bits represent if publisher x publishes symbols 64*y to 64*(y+1) - 1
+    pub publisher_permissions: [[u64; PC_MAX_SYMBOLS_64 as usize]; PC_MAX_PUBLISHERS as usize],
+    pub scores:                [f64; PC_MAX_PUBLISHERS as usize],
+    pub publishers:            [Pubkey; PC_MAX_PUBLISHERS as usize],
+    pub symbols:               [Pubkey; PC_MAX_SYMBOLS as usize],
+}
+
+impl PublisherScoresAccount {
+    fn get_publisher_permission(&self, x: usize, y: usize) -> bool {
+        (self.publisher_permissions[x][y / 64] >> (y % 64)) & 1 == 1
+    }
+
+    fn set_publisher_permission(&mut self, x: usize, y: usize, value: bool) {
+        if value {
+            self.publisher_permissions[x][y / 64] |= 1 << (y % 64);
+        } else {
+            self.publisher_permissions[x][y / 64] &= !(1 << (y % 64));
+        }
```

Comment:
`get_publisher_permission` and `set_publisher_permission` could almost be in their own struct PermissionTable, that we can proptest

Reply:
I will try this at some point to see how much it will change the program size, my current guess is that it will increase it.